pkg/frontend/transport/handler.go

```diff
@@ -230,8 +230,9 @@ func (f *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 			writeError(w, apierror.New(apierror.TypeInternal, err.Error()))
 			return
 		}
-		ctx, _ := context.WithDeadlineCause(r.Context(), deadline,
+		ctx, cancel := context.WithDeadlineCause(r.Context(), deadline,
 			cancellation.NewErrorf("write deadline exceeded (timeout: %v)", f.cfg.ActiveSeriesWriteTimeout))
+		defer cancel()
```

Comment:
Note to reviewers: this HTTP request shouldn't live beyond this function execution, so I think this change is fine.

Reply:
Yep, that's what I saw too.